### PR TITLE
Load user's current intervention/assessment

### PR DESCRIPTION
**Pull request by angelinahan**
This PR adds a few changes related to making sure the user is able to see only their current assessment/intervention

- When a user signs up, an assessment is created with all the words and assigned to them
- We now keep track of a user's current intervention / assessment and load the one they are on
- If a user tries to start to an assessment when they are on an intervention & vice versa, we display an error banner

Screen Shot 2021-02-16 at 6 34 36 PM

### CODE REVIEW

File: frontend/src/firebase/firebaseInteractor.ts [hunk 1]
```diff
@@ -115,10 +124,35 @@ export default class FirebaseInteractor {
...
+  async updateCurrentUser(user: Partial<User>) {
+    this.db.collection("users").doc(this.auth.currentUser?.uid).update(user);
+  }
+
+  async createInitialAssessment() {
+    let words = await this.db.collection("words").get();
+    let wordIds = words.docs.map((word) => word.id);
+    let newAssessmentFields = {
+      currentIndex: 0,
+      // do we need to include in the beginning?
```
Comment (sarah-witt) [inline frontend/src/firebase/firebaseInteractor.ts]: It's not needed but also wouldn't hurt to have for clarity https://firebase.google.com/docs/reference/js/firebase.firestore.FieldValue#static-increment

File: frontend/src/pages/Assessments/data/reducer.ts [hunk 2]
```diff
@@ -24,16 +26,30 @@ const assessmentReducer = (
...
           words: payload.assessment.words,
           firebaseId: payload.assessment.firebaseId,
+          sessionId: payload.assessment.sessionId,
         },
         finished: false,
       };
     case ActionTypes.UPDATE_ASSESSMENT_SUCCESS:
       let { isFinished }: UpdateAssessmentSuccess = payload;
       return {
-        ...state,
```
Comment (sarah-witt) [inline frontend/src/pages/Assessments/data/reducer.ts:33]: why was this removed?

Comment (jrosen081) [reply frontend/src/pages/Assessments/data/reducer.ts:33]: If you are updating the assessment, you should not need the error or the current assessment id again, so we removed it.

File: frontend/src/firebase/firebaseInteractor.ts (hunk 1 above)
Comment (jrosen081) [reply frontend/src/firebase/firebaseInteractor.ts]: Where would this be used?

Comment (sarah-witt) [reply frontend/src/firebase/firebaseInteractor.ts]: Sorry if I wasn't clear- I was responding to the comment in the code. Since we use increment for `durationInSeconds` and increment sets a value if the field does not exist, then a default value of 0 isn't needed when the assessment is created,  but specifying it at assessment creation time wouldn't hurt because it doesn't affect the value and makes it more clear in the code that `durationInSeconds` is an assessment field if that makes sense.

Comment (jrosen081) [reply frontend/src/firebase/firebaseInteractor.ts]: Ah gotcha makes sense. I will keep it for clarity

Comment (sarah-witt) [reply frontend/src/firebase/firebaseInteractor.ts]: Sounds good! Also can you remove or update the comment?

Comment (jrosen081) [reply frontend/src/firebase/firebaseInteractor.ts]: I will remove it

**Review by sarah-witt — APPROVED** [review]
looks good to me!
